tests/test_pandas.py

```diff
@@ -133,17 +135,74 @@ def test_isna() -> None:
-    assert not check(assert_type(pd.notna(pd.NaT), Literal[False]), bool)
-
-    assert check(assert_type(pd.isna(None), Literal[True]), bool)
-    assert not check(assert_type(pd.notna(None), Literal[False]), bool)
-
-    check(assert_type(pd.isna(2.5), bool), bool)
-    check(assert_type(pd.notna(2.5), bool), bool)
+    assert check(assert_type(pd.isna(pd.NA), bool), bool)
+    assert not check(assert_type(pd.notna(pd.NA), bool), bool)
+
+    assert check(assert_type(pd.isna(pd.NaT), bool), bool)
+    assert not check(assert_type(pd.notna(pd.NaT), bool), bool)
+
+    assert check(assert_type(pd.isna(None), bool), bool)
+    assert not check(assert_type(pd.notna(None), bool), bool)
+
+    assert not check(assert_type(pd.isna(2.5), bool), bool)
+    assert check(assert_type(pd.notna(2.5), bool), bool)
+
+    # Check TypeGuard type narrowing functionality
+    # TODO: Due to limitations in TypeGuard spec, the true annotations are not always viable
+    # and as a result the type narrowing does not always work as it intuitively should
+    # There is a proposal being floated for a StrictTypeGuard that will have more rigid narrowing semantics
+    # In the test cases below, a commented out assertion will be included to document the optimal test result
+    nullable1 = random.choice(["value", None, pd.NA, pd.NaT])
+    if pd.notna(nullable1):
+        check(assert_type(nullable1, str), str)
+    if not pd.isna(nullable1):
+        # TODO: This is the true type (see comments above on the limitations of TypeGuard)
+        # check(assert_type(nullable1, str), str)
+        check(assert_type(nullable1, Union[str, NaTType, NAType, None]), str)
+    if pd.isna(nullable1):
+        assert_type(nullable1, Union[NaTType, NAType, None])
+    if not pd.notna(nullable1):
+        # TODO: This is the true type (see comments above on the limitations of TypeGuard)
+        # assert_type(nullable1, Union[NaTType, NAType, None])
+        assert_type(nullable1, Union[str, NaTType, NAType, None])
+
+    nullable2 = random.choice([2, None])
+    if pd.notna(nullable2):
+        check(assert_type(nullable2, int), int)
+    if not pd.isna(nullable2):
+        # TODO: This is the true type (see comments above on the limitations of TypeGuard)
+        # check(assert_type(nullable2, int), int)
+        check(assert_type(nullable2, Union[int, None]), int)
+    if pd.isna(nullable2):
+        # TODO: This is the true type (see comments above on the limitations of TypeGuard)
+        # check(assert_type(nullable2, None), type(None))
+        check(assert_type(nullable2, Union[NaTType, NAType, None]), type(None))
+    if not pd.notna(nullable2):
+        # TODO: This is the true type (see comments above on the limitations of TypeGuard)
+        # check(assert_type(nullable2, None), type(None))
+        assert_type(nullable2, Union[int, None])  # TODO: MyPy result
+        assert_type(
+            nullable2, Union[int, NaTType, NAType, None]
```

Comment:
currently failing here because `nullable2` is `Union[int, None]` (although it should be `None` based on the test of `pd.notna()` )